lwt/ssl_io_dummy.ml

```diff
@@ -0,0 +1,16 @@
+let readf _socket =
+  fun _fd _buffer ->
+  Lwt.fail_with "Ssl not available"
+
+let writev _socket _fd =
+  fun _iovecs ->
+  Lwt.fail_with "Ssl not available"
+
+type client = [ `Ssl_not_available  ]
+type server = [ `Ssl_not_available  ]
```

Comment:
It looks like this mostly solves the issue raised in #92 , by requiring the user to pass in a value of type ``[`Tls_not_available]`` or ``[`Ssl_not_available]`` in order to get one of the runtime failures - could we use this approach in the async PR?  Maybe we could make this a little more avoidable by adding `of Nothing.t` to those types to make them zero-valued?

Reply:
I'm quite certain that #92 has a type like this one.

Reply:
Oh, I see, the problem is that the argument of that type is optional and thus it can be easy to miss the warning in the case that it's missing.

Reply:
That’s a great point. Perhaps we should also fabricate types for certfile and keyfile that won’t typecheck. What do you think?

Additionally, I’m thinking once the mirage adapter in #83 is in, we can reorganize this code to account for the functorization of httpaf-lwt and the `IO` module type.

Reply:
Aren't the certfile and keyfile also optional?  I was thinking maybe do what you're suggesting but with the `Lwt_unix.file_descr` argument, which is required and appears in all four signatures.

Okay, in case of a pending reorganization here it probably makes sense to iron out the interface but not look too hard at the implementation for now.

Reply:
> Aren't the certfile and keyfile also optional?
They are indeed optional. Forgive my confusion.

> I was thinking maybe do what you're suggesting but with the Lwt_unix.file_descr argument
This is a great idea. I really like it

> Okay, in case of a pending reorganization here it probably makes sense to iron out the interface but not look too hard at the implementation for now.
Another discussion we can have is whether this code actually belongs in http/af or not. With the reorganization in #83 - if the changeset is accepted - `httpaf-lwt` becomes a library that provides a functor and a signature for that functor. HTTPS support could well be a third party library that applies that functor with a TLS / SSL `Io` module. That would also reduce the maintenance burden in http/af.

Reply:
Hrmm, I don't do enough public OCaml work to be familiar with the trade-offs of breaking up libraries like that. @seliopou What do you think about the idea of HTTPS support coming in a separate library?